src/core/renderer.ts

```diff
@@ -62,3 +66,55 @@ function copyElementAttributes(destinationElement: Element, sourceElement: Eleme
     destinationElement.setAttribute(name, value)
   }
 }
+
+function createPlaceholderForPermanentElement(permanentElement: PermanentElement) {
```

Comment:
@sstephenson I've merged your latest into this branch, and followed it up by extracting some Placeholder rendering functions to the Renderer class module, then re-used them in the PageRenderer and FrameRenderer.

To help sort out what was shared and what was specific, I've also added `private` function modifiers throughout the `PageRenderer` class.

Are private and exported functions in a Class idiomatic to the Turbo codebase, or would a `renderer/util` module be more appropriate? A class of its own?